snapcraft/storeapi/errors.py

```diff
@@ -632,3 +639,37 @@ class InvalidLoginConfig(StoreError):
 
     def __init__(self, error):
         super().__init__(error=error)
+
+
+def _error_list_to_message(response_json):
+    """Handle error list.
+
+    The error format is given here:
+    https://dashboard.snapcraft.io/docs/api/snap.html#format
+    """
+    messages = []
+    for error_list_item in response_json['error_list']:
+        messages.append(_error_list_item_to_message(
+            error_list_item, response_json))
+
+    return ', and '.join(messages)
+
+
+def _error_list_item_to_message(error_list_item, response_json):
+    """Handle error list item according to code.
+
+    The list of codes are here:
+    https://dashboard.snapcraft.io/docs/api/snap.html#codes
+    """
+    code = error_list_item['code']
+    if code == 'macaroon-permission-required':
+        return _handle_macaroon_permission_required(response_json)
+    else:
+        return error_list_item['message']
+
+
+def _handle_macaroon_permission_required(response_json):
+    if response_json['permission'] == 'channel':
+        channels = response_json['channels']
+        return 'Lacking permission to release to channel(s) {}'.format(
+            formatting_utils.humanize_list(channels, 'and'))
```

Comment:
Can we have an `else:` branch here to make this more robust?

Reply:
Yeah, let me rework this a little. Never quite sure what the API is gonna give us.

Reply:
Alright, it's more robust now, it'll just fall back to the default (typically unhelpful) message.